### PR TITLE
feat: kosmosctl image pull/push

### LINKED ISSUE #109: kosmosctl command should support offline installation
**What would you like to be added**:
kosmosctl command can support offline installation

**Why is this needed**:
`kosmosctl image pull --image-list=[image-list.txt] --output=[output-dir] --container-runtime=[container-runtime]`
`kosmoscrl image push --image-list=[image-list.txt] --artifact=[*.tar.gz] --username=[registry-username] --password=[registry-password] --private-registry=[private-registry-name]
`

### PR DESCRIPTION
**Pull request by kuangcheng66**
feat: kosmosctl image pull/push

#### What type of PR is this?
- **feat: add kosmosctl image pull/push func** 

#### What does this PR do?
`kosmosctl image pull --kosmos-version=[default: latest] --coredns-version=[default: latest] --eps-version=[default: latest]`
`kosmoscrl image push --artifact=[*.tar.gz] --username=[registry-username] --private-registry=[private-registry-name]`
#### Which issue(s) does this PR fix?

Fixes #109

### CODE REVIEW

File: pkg/kosmosctl/image/pull.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+	return nil
+}
+
+func (o *CommandPullOptions) Validate() error {
+	return nil
+}
+
+func (o *CommandPullOptions) Run() error {
+	// 1. pull image from public registry
+	klog.Info("Start pulling images ...")
```
Comment (wuyingjun-lucky) [inline pkg/kosmosctl/image/pull.go]: I think we may set the log level

File: pkg/kosmosctl/image/pull.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+	err = o.ImageSave(imageList)
+	if err != nil {
+		klog.Infof("image save failed: %s", err)
+		return err
+	}
+	klog.Info("kosmos-io.tar.gz has been saved successfully. ")
+	return nil
+}
+
+func (o *CommandPullOptions) ImagePull() (imageList []string, err error) {
```
Comment (wuyingjun-lucky) [inline pkg/kosmosctl/image/pull.go]: ImagePull ---> pullImage

File: pkg/utils/constants.go [hunk 3]
```diff
@@ -7,6 +7,39 @@ import (
...
+		ClusterLinkControllerManager,
+		ClusterLinkProxy,
+		ClusterLinkElector,
+		ClusterLinkFloater,
+		Coredns,
+		EpsProbePlugin}
+)
+
+const (
+	ClusterTreeKnodeManager      = "ghcr.io/kosmos-io/clustertree-knode-manager"
```
Comment (wuyingjun-lucky) [inline pkg/utils/constants.go]: I think the KnodeManager is discard

File: pkg/kosmosctl/image/pull.go (hunk 1 above)
Comment (kuangcheng66) [reply pkg/kosmosctl/image/pull.go]: Okay, I'll set the output level of the log on the entire code

File: pkg/kosmosctl/image/pull.go (hunk 2 above)
Comment (kuangcheng66) [reply pkg/kosmosctl/image/pull.go]: ok

File: pkg/utils/constants.go (hunk 3 above)
Comment (kuangcheng66) [reply pkg/utils/constants.go]: Okay, I'll recheck the names of all the images.

**Pull request merged**